docs: More documentation cleanup (#2106)
DaveSkender/Stock.Indicators: docs/migration.md

Author: ```suggestion
```

Author: This shouldn’t be correct, review section

Author: This shouldn’t be correct, review section

Author: ```suggestion
IReadOnlyList<SmaResult> results = bars.ToSma(20);
```

Many places

Author: This should merge into doc site and just be succinct highlight here, or have a separate “What’s new in v3”

Popular indicators with complete streaming documentation:

- Moving Averages: [SMA](/indicators/sma), [EMA](/indicators/ema), [WMA](/indicators/wma)
- Oscillators: [RSI](/indicators/rsi), [MACD](/indicators/macd), [Stochastic](/indicators/stoch)
- Channels: [Bollinger Bands](/indicators/bollinger-bands), [Keltner](/indicators/keltner)

## Known issues and tips

> A few v3 behaviors commonly trip up migrating code. Review these before reporting an issue.

Author: ```suggestion
A few v3 known sub-optimal behaviors worth noting before reporting an issue. These are in our backlog.
```

// BufferList: size the list
SmaList smaList = new(20) { MaxListSize = 250_000 };
```

`maxCacheSize` is inherited by every chained hub, so size it for the **largest warmup in the whole chain**. Don't shrink it down to a single indicator's lookback — each hub validates the size against its own warmup floor at construction and throws `ArgumentOutOfRangeException` if it is too small. See [Stream hub memory management](/guide/styles/stream#memory-management) for details.
:::

### Corrections after pruning are approximate

Once the cache has pruned old bars, a late arrival or correction that triggers a rebuild can no longer see the pruned history, so the rebuilt values for stateful indicators won't exactly match a hub that received the same data in order. Keep the cache large enough to cover your expected late-arrival window.

### Mutate the root hub only

Feed and correct streaming data through the `BarHub` (or `TradeTickHub`) you created — it cascades to every dependent hub. Calling `Add`, `RemoveAt`, `RemoveRange`, `Remove`, or `Reinitialize` on a subscribed hub such as an `EmaHub` throws `InvalidOperationException`. See the [streaming guide](/guide/styles/stream#thread-safety).

> ### `null` versus `NaN` results

Author: This shouldn’t be correct
